File: Napakalaki RUBY/lib/treasure.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  class Treasure
  
  def initialize(n,bonus,t)
    @name = n
    @bonus = bonus
    @type = t
  end
  
  def getName()
    @name
  end
  
  def getBonus()
    @bonus
  end
  
  def getType()
    @type
  end
  
  def to_s
    return "\n Nombre: #{@name} \n Bonus: #{@bonus}\n Tipo: #{@type}"
  end
end

end

File: Napakalaki JAVA/src/napakalaki_java/Treasure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki_java;

/**
 *
 * @author jumacasni
 */
public class Treasure {
    private String name;
    private int bonus;
    private TreasureKind type;
    
    public Treasure(String n, int bonus, TreasureKind t){
        name = n;
        this.bonus = bonus;
        type = t;
    }
    
    public String getName(){
        return name;
    }
    
    public int getBonus(){
        return bonus;
    }
    
    public TreasureKind getType(){
        return type;
    }    
    
    public String toString(){
        String s = "";
        s = "Nombre: " + name + "\nBonus: " + bonus + "\nTipo: " + type;
        return s;
    }
}
